wallet/src/account/output_cache/mod.rs

```diff
@@ -91,17 +98,44 @@ impl OutputCache {
         }
     }
 
-    fn valid_utxo(&self, outpoint: &UtxoOutPoint) -> bool {
+    fn valid_utxo(
+        &self,
+        outpoint: &UtxoOutPoint,
+        output: &TxOutput,
+        transaction_block_info: &BlockInfo,
+        current_block_info: &BlockInfo,
+    ) -> bool {
         !self.consumed.contains(outpoint)
+            && output.timelock().map_or(true, |timelock| match timelock {
```

Comment:
Have we already considered reusing checks from tx_verifier? Not for this PR, but in general we might want to open an issue for that to look later.